GameServer/User.cpp

```diff
@@ -1199,27 +1199,49 @@ void CUser::SendPremiumInfo()
 */
 void CUser::RequestUserIn(Packet & pkt)
 {
+	uint16 sReqUserCount = pkt.read<uint16>();
+
+	if (sReqUserCount > 1000)
+		sReqUserCount = 1000;
+
 	Packet result(WIZ_REQ_USERIN);
-	short user_count = pkt.read<uint16>(), online_count = 0;
-	if (user_count > 1000)
-		user_count = 1000;
+	size_t wpos = result.wpos();
 
-	result << uint16(0); // placeholder for user count
+	uint16 sUserCount = 0;
+	result << sUserCount;
 
-	for (int i = 0; i < user_count; i++)
+	for (int i = 0; i < sReqUserCount; i++)
 	{
 		CUser *pUser = g_pMain->GetUserPtr(pkt.read<uint16>());
-		if (pUser == nullptr || !pUser->isInGame())
+		if (pUser == nullptr
+			|| pUser == this
+			|| !pUser->isInGame()
+			|| pUser->isGM()
+			|| pUser->GetZoneID() != GetZoneID())
```

Comment:
Can you please add a comment of why GM's and users from the same ZoneID should not be considered in RequestUserIn handler? In what scenario will it be the case?